qtranslate_core.php

```diff
@@ -1118,20 +1118,6 @@ function qtranxf_convertURL( $url = '', $lang = '', $forceadmin = false, $showDe
     if ( empty( $lang ) ) {
         $lang = $q_config['language'];
     }
-    if ( empty( $url ) ) {
-        // TODO refactor this hack for qtranslate-slug! We might need a hook here.
-        if ( $q_config['url_info']['doing_front_end'] && function_exists( 'qts_get_url' ) && $q_config['url_mode'] != QTX_URL_QUERY ) {
-            // quick workaround, but need a permanent solution
-            $url = qts_get_url( $lang );
-            if ( ! empty( $url ) ) {
```

Comment:
For what I've understood of original intent of this part, this is just to handle links to current page (e.g. language switcher), which is the case you need a link in another language to be shown.
Not sure how `qts_get_url `could return empty string, I guess that condition is only given conservatively to avoid infinite loop, as `qtranxf_convertURL `was called recursively. This is superseded with the filter.